### PR TITLE
[stable/jenkins] Make Jenkins master UI test optional

### PR DESCRIPTION
**Pull request by siwyd**
#### What this PR does / why we need it:

The UI test makes it difficult to do an idempotent deployment check
because the randomly named UI test Pod will always want to deploy
a new pod. For those that don't need the UI test but prefer an
idempotent deploy, allowing disabling the UI test seems like a good
idea to me. This commit does the following:

- Allow disabling the master UI test (skip test Pod + ConfigMap)
- Bump chart version to 1.5.10

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by torstenwalter** [comment]
@siwyd How do you install the chart? If you are using ` helm install stable/jenkins --version 1.6.0` the test pod is not created, only the configmap for the tests.

**Comment by siwyd** [comment]
@torstenwalter I don't use tiller server side component, I use helm template and pipe the output to kapp to deploy. This is the Pod resource in question: https://github.com/helm/charts/blob/master/stable/jenkins/templates/tests/jenkins-test.yaml. Is there a difference between install and template then? Does install not deploy Pod resources for some reason? 

My PR was based on 1.5.9 first, but 1.6.0 still has the problem for me:
```
❯ k get po --namespace jenkins
NAME                       READY   STATUS    RESTARTS   AGE
jenkins-5d8447b886-s5n22   1/1     Running   0          2m43s
jenkins-ui-test-vmn1t      0/1     Error     0          2m45s
```
It also always errors out for some reason, but that doesn't actually matter to me. The non-idempotency of the Pod is the problem for my deployment flow.

**Comment by torstenwalter** [comment]
@siwyd `helm install` does not install test pods https://helm.sh/docs/developing_charts/#chart-tests

File: stable/jenkins/README.md [hunk 1]
```diff
@@ -163,6 +163,7 @@ The following tables list the configurable parameters of the Jenkins chart and t
 | `master.prometheus.alertingrules` | Array of prometheus alerting rules | `[]`                                        |
 | `master.prometheus.alertingRulesAdditionalLabels` | Additional labels to add to the prometheus rule object     | `{}`                                   |
 | `master.priorityClassName`        | The name of a `priorityClass` to apply to the master pod | Not set               |
+| `master.test.enabled`             | Enable master UI test during deployment | `true`                                 |
```
Comment (torstenwalter) [inline stable/jenkins/README.md]: Should we rename the description to something like "Can be used to disable rendering test resources when using helm template"?

File: stable/jenkins/values.yaml [hunk 2]
```diff
@@ -352,6 +352,10 @@ master:
     # The `groups` root object is added by default, simply add the rule entries
     alertingrules: []
 
+  # Jenkins master UI test
+  test:
```
Comment (torstenwalter) [inline stable/jenkins/values.yaml]: we should avoid nesting here to follow best practices 

> In most cases, flat should be favored over nested. The reason for this is that it is simpler for template developers and users. https://helm.sh/docs/chart_best_practices/#flat-or-nested-values

> Variables names should begin with a lowercase letter, and words should be separated with camelcase https://helm.sh/docs/chart_best_practices/#naming-conventions

File: stable/jenkins/README.md (hunk 1 above)
Comment (siwyd) [reply stable/jenkins/README.md]: Alright, sounds good to me, I'll tweak this.

File: stable/jenkins/values.yaml (hunk 2 above)
Comment (siwyd) [reply stable/jenkins/values.yaml]: Alright!

Comment (siwyd) [reply stable/jenkins/values.yaml]: @torstenwalter Pushed change. Did you mean to unnest completely? So use 'master.test.enabled'? I just changed it to test.enabled. The whole chart is nested so I don't wanna make it look too out of line with everything else?

File: stable/jenkins/README.md (hunk 1 above)
Comment (siwyd) [reply stable/jenkins/README.md]: @torstenwalter Pushed change.

File: stable/jenkins/values.yaml (hunk 2 above)
Comment (torstenwalter) [reply stable/jenkins/values.yaml]: I thought about `master.enableTest`. We check for presence of master key anyhow. If you introduce test and below that enabled then you would also need to cater for if master.test is present or not.

File: stable/jenkins/README.md (hunk 1 above)
Comment (siwyd) [reply stable/jenkins/README.md]: @torstenwalter Apologies for the delay. I pushed the changes for you to review.

**Comment by torstenwalter** [comment]
@siwyd Thank you for the PR.

**Comment by siwyd** [comment]
@torstenwalter Thx for the quick feedback.

**Pull request merged**